### ISSUE TITLE
win_security_policy cannot find secedit.ini from exported tmp file

### ISSUE THREAD

**Issue by GingerGraham** [ansible-collections/community.windows#348]
##### SUMMARY
Error trying to import the exported secedit.ini while trying to set Windows Security Policy using win_security_policy

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
win_security_policy 1.8.0

##### ANSIBLE VERSION

```
ansible [core 2.11.6] 
  config file = /Users/grwatts/Development/Vocera/Bitbucket/vocera-tooling-devops/ansible/ansible.cfg
  configured module search path = ['/Users/grwatts/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/4.8.0/libexec/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/grwatts/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.0 (default, Oct 22 2021, 13:39:57) [Clang 13.0.0 (clang-1300.0.29.3)]
  jinja version = 3.0.2
  libyaml = True
```

##### COLLECTION VERSION
```
$ ansible-galaxy collection list community.windows

Collection        Version
----------------- -------
community.windows 1.8.0  
```

##### CONFIGURATION
```
ANSIBLE_PIPELINING(/Users/grwatts/Development/Vocera/Bitbucket/vocera-tooling-devops/ansible/ansible.cfg) = True
DEFAULT_VAULT_PASSWORD_FILE(/Users/grwatts/Development/Vocera/Bitbucket/vocera-tooling-devops/ansible/ansible.cfg) = /Users/grwatts/Development/Vocera/Bitbucket/vocera-tooling-devops/ansible/vaultpass
HOST_KEY_CHECKING(/Users/grwatts/Development/Vocera/Bitbucket/vocera-tooling-devops/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT
Windows Server 2019
1809 Build 17763.2300

##### STEPS TO REPRODUCE
1. Standup vanilla Windows Server 2019 image (on AWS from AWS AMI in this test case)
2. Configure local user and CredSSP for ansible to run against the system
3. Run below play as part of a playbook

```yaml
- name: 'Configure Local Security Policy - Max Password Age'
  community.windows.win_security_policy:
    section: System Access
    key: MaximumPasswordAge
    value: '{{system_access.max_password_age}}' # Variable set to 0 to disable this policy
```

##### EXPECTED RESULTS
I would expect the Windows Security Policy for Max Password Age to be set to the value passed as a variable, in this case 0

##### ACTUAL RESULTS
- Module returns that it cannot find the exported tmp file.
- Inspecting the location on the file system shows a tmp file with the same name except the last character is incremented by one (hex)

```paste below
TASK [win_rds : Configure Local Security Policy - Max Password Age] **********************************************************************************************************************************************************
task path: /Users/grwatts/Development/Vocera/Bitbucket/vocera-tooling-devops/ansible/roles/win_rds/tasks/main.yml:28
Using module file /Users/grwatts/.ansible/collections/ansible_collections/community/windows/plugins/modules/win_security_policy.ps1
Pipelining is enabled.
<172.30.212.221> ESTABLISH WINRM CONNECTION FOR USER: ansible on PORT 5986 TO 172.30.212.221
EXEC (via pipeline wrapper)
fatal: [172.30.212.221]: FAILED! => {
    "changed": true,
    "import_log": "Completed 1 percent (0/63) \tProcess Privilege Rights area        \n\nCompleted 3 percent (1/63) \tProcess Privilege Rights area        \n\nCompleted 4 percent (2/63) \tProcess Privilege Rights area        \n\nCompleted 6 percent (3/63) \tProcess Privilege Rights area        \n\nCompleted 7 percent (4/63) \tProcess Privilege Rights area        \n\nCompleted 9 percent (5/63) \tProcess Privilege Rights area        \n\nCompleted 11 percent (6/63) \tProcess Privilege Rights area        \n\nCompleted 12 percent (7/63) \tProcess Privilege Rights area        \n\nCompleted 14 percent (8/63) \tProcess Privilege Rights area        \n\nCompleted 15 percent (9/63) \tProcess Privilege Rights area        \n\nCompleted 17 percent (10/63) \tProcess Privilege Rights area        \n\nCompleted 19 percent (11/63) \tProcess Privilege Rights area        \n\nCompleted 20 percent (12/63) \tProcess Privilege Rights area        \n\nCompleted 25 percent (15/63) \tProcess Privilege Rights area        \n\nCompleted 25 percent (15/63) \tProcess Group Membership area        \n\nCompleted 49 percent (30/63) \tProcess Group Membership area        \n\nCompleted 49 percent (30/63) \tProcess Registry Keys area        \n\nCompleted 49 percent (30/63) \tProcess File Security area        \n\nCompleted 49 percent (30/63) \tProcess Services area        \n\nCompleted 65 percent (40/63) \tProcess Services area        \n\nCompleted 73 percent (45/63) \tProcess Services area        \n\nCompleted 73 percent (45/63) \tProcess Security Policy area        \n\nCompleted 77 percent (48/63) \tProcess Security Policy area        \n\nCompleted 84 percent (52/63) \tProcess Security Policy area        \n\nCompleted 88 percent (55/63) \tProcess Security Policy area        \n\nCompleted 93 percent (58/63) \tProcess Security Policy area        \n\nCompleted 100 percent (63/63) \tProcess Security Policy area        \n\n                                                                           \nThe parameter is incorrect.\n\nThe task has completed with an error.",
    "key": "MaximumPasswordAge",
    "msg": "Failed to import secedit.ini file from C:\\Users\\ansible\\AppData\\Local\\Temp\\tmpB8A9.tmp",
    "rc": 1,
    "section": "System Access",
    "stderr": null,
    "stderr_lines": [],
    "stdout": "",
    "stdout_lines": [],
    "value": "0"
}

```
![image](https://user-images.githubusercontent.com/34165628/146366517-72c1fef7-4d0f-42b5-88fb-6ec6b8e43009.png)

**Comment by jborean93** [comment]
You code isn't failing because it cannot find the file, it's failing becaue `SecEdit.exe` is reporting an error, albiet a very vague one. The msg is just a generic boilerplate that says it failed to import the file and `import_log` contains a bit a more detail in it

```
Completed 1 percent (0/63) 	Process Privilege Rights area        

---SNIP---

Completed 100 percent (63/63) 	Process Security Policy area        

                                                                           
The parameter is incorrect.

The task has completed with an error.
```

The code that is running this is https://github.com/ansible-collections/community.windows/blob/65ddb7180c737e7a43734697c06cf6de0d09c3e6/plugins/modules/win_security_policy.ps1#L80-L114 and essentially it will create the temp files but also delete them once no longer needed before the error is reported. This is why you cannot see the file on your system. I'm not sure what the other temp files are but I would guess they are unrelated or at least not managed by Ansible.

Ultimately you need to figure out why it's rejecting the value, a quick search online suggests maybe using `-1` instead of `0` for this value https://community.spiceworks.com/topic/2257912-cannot-update-password-complexity-on-hyper-v-server-2019-core https://github.com/dsccommunity/SecurityPolicyDsc/issues/121. The docs for this setting https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-gpsb/0b40db09-d95d-40a6-8467-32aedec8140c also indicate that it needs to be `-1` to avoid conflicting with `MinimumPasswordAge`

> Maximum number of days that a password can be used  before the client requires  the user to change it. The value MUST be either equal to "-1" or in  the range 1 to 999. The value "-1" indicates that a password never  expires. If the maximum password age value is not "-1", the minimum password age MUST be less than the maximum password age.

Essentially if it's not `-1` then the value needs to be more than the minimum password age.

**Comment by GingerGraham** [comment]
Thanks @jborean93 !

As it turns out, it's exactly as you found, simply changing the MaxPasswordAge to -1 resolves the error.

2 things threw me here:

1. When configured in the UI the value set for MaxPasswordAge is 0 not -1
2. The filename offset thing

While trying to workaround this I had noticed that secedit showed -1 in the database rather than 0, I just never joined the dots.  Shame the MS Docs for secedit itself don't lead to the rather more useful page you found!

Anyway, mea culpa, thank you for debugging my bad code, I'm glad there's not an issue in the plugin itself.

Interestingly, the files I highlighted only get created during the ansible play runtime.  So they must be linked in some way; just not sure now exactly.  

Problem solved... Thank you!
